_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-videopress.php

```diff
@@ -100,6 +124,36 @@ public function videopress_block_update_meta( $request ) {
 
 		$response_body = json_decode( wp_remote_retrieve_body( $result ) );
 		if ( is_bool( $response_body ) && $response_body ) {
+
+			// VideoPress data is stored in attachment meta for Jetpack sites, but not on wpcom.
+			if ( ! defined( 'IS_WPCOM' ) || ! IS_WPCOM ) {
+				$meta               = wp_get_attachment_metadata( $post_id );
+				$should_update_meta = false;
+
+				if ( ! $meta ) {
+					return rest_ensure_response(
+						new WP_Error(
+							'error',
+							__( 'Attachment meta was not found.', 'jetpack' )
+						)
+					);
+				}
+
+				if ( isset( $json_params['display_embed'] ) && isset( $meta['videopress']['display_embed'] ) ) {
+					$meta['videopress']['display_embed'] = $json_params['display_embed'];
+					$should_update_meta                  = true;
+				}
+
+				if ( isset( $json_params['rating'] ) && isset( $meta['videopress']['rating'] ) ) {
+					$meta['videopress']['rating'] = $json_params['rating'];
```

Comment:
I think we should add some validation here, so one doesn't just pass random parameters that won't really mean much.

Reply:
Good call. The endpoint it calls prior to saving does validate the value, but if its ever changed to ignore the value instead of returning a failed response this could still save when it shouldn't.

I've refactored some existing rating text validation into a helper function so we're not duplicating the rating strings everywhere.